### PR TITLE
MM-34093 Adding English Australia to the languages

### PR DESCRIPTION
**Pull request by ctlaltdieliet**
Adding English Australia to the languages

### CODE REVIEW

File: app/i18n/index.js
Comment (enahum) [inline app/i18n/index.js:31]: We don't support language variants just yet

**Review by enahum — CHANGES_REQUESTED** [review]
Also languages should be included by our translation server

File: app/i18n/index.js
Comment (ctlaltdieliet) [reply app/i18n/index.js:31]: I thought we do (Chinese Traditional and Chinese Simplified are both in shipped)
The language files for English Australian are in Weblate and are added in a different PR @metanerd ?

Comment (metanerd) [reply app/i18n/index.js:31]: We could include the empty JSON file in the appropriate path. 
en-AU.json:
```
{}
```
And on Monday the real translations would be added via weblate PR. Then the code won't break, because it doesn't find the file.

Comment (enahum) [reply app/i18n/index.js:31]: Then maybe merge this when the translation has been added and PR updated with the latest master containing the translation file.

Also default device language selector is using `languageCode` instead of `languageTag` so no country variant is being picked up until user logs in. If we are going to change this as well it should be done in a way that for those variants it applies but for unsupported variants it should fallback to `languageCode` instead. (See app/selectors/i18n)

Comment (metanerd) [reply app/i18n/index.js:31]: Ah, I see the problem now. So language variation is blocked until we implement the device language selector using `languageTag` with fallback `languageCode`. If we don't do this until Monday, we cannot add the language file.

Comment (ctlaltdieliet) [reply app/i18n/index.js:31]: I'm new to the code but is this line doing what I think it does: overriding it for 3 languages? Can we add en-AU there?
https://github.com/mattermost/mattermost-mobile/blob/48d4775907d03c3edf2cd7e6610c6f194ee02465/test/setup.js#L186

Comment (enahum) [reply app/i18n/index.js:31]: The variation is going to apply for those logged in users that have set the language variation in they profile and will fallback to regular english (the same happens with the three variations we have "support" today) unless we make the needed changed in the selector found in `app/selectors/i18n`.

Even if we decide that we don't care about non logged in users we need to test this PR, in order for us to be able to test it with some level of accuracy the translation file must be present. That means that this PR needs to be rebased after the translation PR is merged.

Not sure if I'm explaining myself

Comment (jasonblais) [reply app/i18n/index.js:31]: @enahum do you just mean that we need the `en-AU.json` file submitted via Weblate on Monday and merged to https://github.com/mattermost/mattermost-mobile/tree/master/assets/base/i18n first, then we can review and merge this PR?

cc @ctlaltdieliet @metanerd

**Comment by ctlaltdieliet** [comment]
Hi @josephbaylon,
can you make the necessary changes to languageTag? 
I'm not a skilled enough in mobile development to make the changes myself.

File: app/i18n/index.js
Comment (jasonblais) [reply app/i18n/index.js:31]: @enahum @ctlaltdieliet Instead of shipping it as a language variant, is there an option to ship English-Australian as a separate language, similar to how we ship the two variations of Chinese as separate languages?

Then in that case, I believe we need to submit a PR to https://github.com/mattermost/mattermost-mobile/tree/master/assets/base/i18n first (via Weblate), then update this file appropriately.

Let me know if I'm misunderstanding.

**Review by enahum — APPROVED** [review]
Waiting gor the othe two PR's

**Comment by enahum** [comment]
Closing in favor of https://github.com/mattermost/mattermost-mobile/pull/5411

**Pull request closed without merging**
